### ISSUE TITLE
Support handling messages from other bots

### ISSUE THREAD

**Issue by aschokking** [llimllib/limbo#44]
We have a bunch of scenarios where we would like to monitor the output of other integrations (like jenkins build notifications) and it looks like out of the box limbo is filtering those out from ever reaching the plugins. I can understand scenarios where this might not be desirable, maybe it could be a config option?

**Comment by llimllib** [comment]
For sure, I'd love to see that.

I'd guess those come in as messages from user "slackbot"? If so, they get filtered out [here](https://github.com/llimllib/limbo/blob/master/limbo/limbo.py#L106). Are you interested in developing this?

**Comment by aschokking** [comment]
I think each integration gets it's own username, but in principal I had to delete that whole function to get things to work :) Sure would be happy to take a crack at a patch. Does the config option sound reasonable? Were there concrete scenarios for filtering out bots or is this something we could just plain open up in general?

**Comment by llimllib** [comment]
The filtering is only designed to prevent feedback loops, so I'd be happy for all bot messages to get through.

**Comment by llimllib** [comment]
testing

**Comment by llimllib** [comment]
ah it's actually [this line](https://github.com/llimllib/limbo/blob/master/limbo/limbo.py#L94) that filters the bot messages.

**Comment by llimllib** [comment]
the problem is that the message has no "user". Here's what limbo gets from the slack API on a github integration message:

```
{
  'subtype': 'bot_message',
  'type': 'message',
  'ts': '1432905377.000011',
  'channel': 'C02L9JD16',
  'bot_id': 'B03QW74AF',
  'attachments': [{
    'fallback': '[llimllib/limbo] New comment on issue <https://github.com/llimllib/limbo/issues/42#issuecomment-106799926|#42: add flickr creative commons search plugin>',
    'id': 1,
    'title': 'Comment by llimllib\n',
    'pretext': '[llimllib/limbo] New comment on issue <https://github.com/llimllib/limbo/issues/42#issuecomment-106799926|#42: add flickr creative commons search plugin>',
    'text': 'test',
    'color': 'FAD5A1',
    'mrkdwn_in': ['text', 'pretext']
  }],
  'text': ''
}
```

**Comment by llimllib** [comment]
I've filed a help request with slack to find out if it's possible to go from the bot_id to the name of the bot, will keep you updated on the results.

**Comment by aschokking** [comment]
Sounds great, thanks for digging in. Yeah I ran into the no user thing, part of the reason I needed to just empty the block to get it to work locally.

**Comment by llimllib** [comment]
How did you get your code to work when you weren't sure what integration was sending the message?

**Comment by aschokking** [comment]
I just let everything go through and knew only the integration would be sending the particular text pattern I was looking for (although it was nice being able to send it myself for testing as well)

**Comment by llimllib** [comment]
Turns out the integration info is sent to the slack client on connect, but python-slackclient doesn't save it for you in any convenient way. Will update it to do so.

**Comment by llimllib** [comment]
btw, I'm not ignoring you, I've [forked slack's client library](https://github.com/llimllib/slackrtm) to add bot handling to it. Should be done soon!

**Comment by aschokking** [comment]
No worries! We hacked around it so not blocked but I would be happier with stock version of limbo instead. Thanks for working on the comprehensive fix!

**Comment by llimllib** [comment]
The fix for this was finally merged on Sep 23 in aac0e69a777632b377068d7677bb91678dbf79f8. Thanks for the report!
